### ISSUE TITLE
bug: flaky tests 🔨

### ISSUE THREAD

**Issue by EvolveArt** [keep-starknet-strange/madara#428]
# Bug Report

**Madara version:** 

https://github.com/keep-starknet-strange/madara/commit/3db1adafd06259090a2df80cae6617a904b318c3

**Current behavior:**

There are some flaky integration tests.

`Get transaction by blockId and index`

**Expected behavior:**

Tests should always pass.

**Steps to reproduce:**

Run the integration tests with `npm run test-seq` in the `tests/` folder.

**Related code:**

We get the following error :

```
LibraryError: -32601: Method not found
```

**Other information:**

**Comment by ClementWalter** [comment]
I do have errors locally on main branch but not the same ones
<img width="635" alt="image" src="https://github.com/keep-starknet-strange/madara/assets/18620296/d5dd2fb8-f670-458e-9697-f971620e6545">

**Comment by ClementWalter** [comment]
indeed, see related job https://github.com/keep-starknet-strange/madara/actions/runs/5043252152/jobs/9045546080

**Comment by ClementWalter** [comment]
but I'm also on latest main, how come we (I) don't have the same error?

**Comment by ftupas** [comment]
> but I'm also on latest main, how come we (I) don't have the same error?

Not sure why, I'm on node v20.2.0 and did `npm i` before running the test, that's a bit strange we have different errors 😨

**Comment by ClementWalter** [comment]
I have 
```
$ npm install

up to date, audited 992 packages in 10s
```
```
$ node -v
v20.0.0
```
and `git log`
```
commit 3db1adafd06259090a2df80cae6617a904b318c3 (HEAD -> main, madara/main)
Author: Abdel @ StarkWare <45264458+abdelhamidbakhta@users.noreply.github.com>
Date:   Mon May 22 09:58:13 2023 +0200

    📝 Add doc badges (#425)
```

**Comment by ftupas** [comment]
I managed to replicate this, I did `cargo build --release` and ran the tests `npm run test-seq` afterwards
![image](https://github.com/keep-starknet-strange/madara/assets/35031356/e717c93e-eeef-4ddf-b7ad-bb1593582ed8)


> I do have errors locally on main branch but not the same ones <img alt="image" width="635" src="https://user-images.githubusercontent.com/18620296/239887332-d5dd2fb8-f670-458e-9697-f971620e6545.png">

**Comment by sparqet** [comment]
That's weird, I have every test working on my machine

<img width="1132" alt="Capture d’écran 2023-05-23 à 16 09 46" src="https://github.com/keep-starknet-strange/madara/assets/37338401/e87428cc-57b4-4741-9846-d9557a258922">

**Comment by ftupas** [comment]
> That's weird, I have every test working on my machine
> <img alt="Capture d’écran 2023-05-23 à 16 09 46" width="1132" src="https://user-images.githubusercontent.com/37338401/240307853-e87428cc-57b4-4741-9846-d9557a258922.png">

I think this can be closed now, @greged93 did a fix with #435  💪

**Comment by greged93** [comment]
I think we have two separate issues: the above failing tests mentioned by @ClementWalter and @ftupas which were fixed with #435, but we still have some flaky CI tests (see [here](https://github.com/keep-starknet-strange/madara/actions/runs/5056174261/jobs/9073310249)). So for me this remains open.

**Comment by ftupas** [comment]
> I think we have two separate issues: the above failing tests mentioned by @ClementWalter and @ftupas which were fixed with #435, but we still have some flaky CI tests (see [here](https://github.com/keep-starknet-strange/madara/actions/runs/5056174261/jobs/9073310249)). So for me this remains open.

Ah gotcha! Thanks sir 🫡
